source/lib/repositories-service.js

```diff
@@ -27,18 +29,17 @@ export async function getRepositories(
 export async function listRepositories(update) {
 	const stored = await repositoriesStorage.getAll();
 
-	let tree = stored;
+	const tree = stored;
 	if (update || !tree || Object.keys(tree).length <= 0) {
 		const fetched = await getRepositories();
 		/* eslint-disable camelcase */
-		tree = fetched.reduce((tree, {full_name}) => {
+		for (const {full_name} of fetched) {
 			const {owner, repository} = parseFullName(full_name);
-			return Object.assign({}, tree, {
-				[owner]: Object.assign(tree[owner] || {}, {
-					[repository]: Boolean(stored && stored[owner] && stored[owner][repository])
-				})
+			tree[owner] = Object.assign(tree[owner] || {}, {
+				[repository]: Boolean(stored && stored[owner] && stored[owner][repository])
 			});
```

Comment:
.reduce brought us to do unspeakable things

```suggestion
			tree[owner] = tree[owner] || {};
			tree[owner][repository] = Boolean(stored && stored[owner] && stored[owner][repository]);
```

Or this?

```suggestion
			tree[owner] = tree[owner] || {};
			tree[owner][repository] = Boolean(stored?.[owner]?.[repository]);
```

Reading the code, however, it seems that this code is wrong in many parts. `stored` is never null nor empty since it gets the default options.

Reply:
The latter

Reply:
I'll have to pick that in the next PR because this branch is Node 11+